Submodule update documentation

dhaval0894 opened the pull request:
<!-- Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request. -->


**Detailed description**

<!-- Explain the **details** for making this change. What existing problem does the pull request solve? Please provide enough information so that others can review your pull request -->
This pull request adds the documentation for updating all the submodules or a specific submodule in the repo. 
![documentation](https://user-images.githubusercontent.com/7266024/75289635-ffaccf80-581e-11ea-83b3-7d65e5aca026.png)

**Test plan (required)**

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

<!-- **Code formatting**
Make sure you ran astyle on your code before making the PR. Check our contribution guidelines here: https://cutter.re/docs/code.html -->
Dev Documentation.

**Closing issues**

<!-- put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such). -->
Closes #2066

ITAYC0HEN commented on docs/source/building.rst:
```diff
@@ -70,6 +70,24 @@ that you can start like this:
 
    ./build/Cutter
 
+Make sure to update the submodule whenever the submodule upstream is updated. 
```
Here is a suggestion for the opening sentence and a section title. The entire change should be added to the bottom of `code.rst` so don't approve the suggestions and commits here.

```suggestion
Updating the submodules
~~~~~~~~~~~~~~~~~~~~~~~
Git submodules play a major part in Cutter. This, because Cutter is powered by radare2, its parent project, and tries to stay up-to-date with its recent version, which allows us to implement new features, and enjoy bug fixes and performance improvements on radre2. Often, we need to update the radare2 submodule or others, in order to push the most recent version of them to Cutter.
```

ITAYC0HEN commented on docs/source/building.rst:
```diff
@@ -70,6 +70,24 @@ that you can start like this:
    ./build/Cutter
 
+Make sure to update the submodule whenever the submodule upstream is updated. 
+You can update all the submodules at once by running these commands from the project root:
```
I'll add instructions on how to list all available submodules from the cutrer root folder. e.g `git config --file .gitmodules --get-regexp path | awk '{ print $2 }'`

ITAYC0HEN commented on docs/source/building.rst:
```diff
@@ -70,6 +70,24 @@ that you can start like this:
+   git add submodule_name_1 submodule_name_2
+   git commit -m "Update submodules"
+
+Alternatively, you can also update the submodules individually
```
```suggestion
More likely, you'll only need to update the radare2 submodule. In order to update one submodule individually, use the following code:
```

ITAYC0HEN commented on docs/source/building.rst:
```diff
@@ -70,6 +70,24 @@ that you can start like this:
+   git checkout master && git pull
+   cd ..
+   git add radare2
+   git commit -m "Update radare2"
```
```suggestion
   git commit -m "Update radare2 submodule"
```

dhaval0894 commented:
Thanks a lot for taking the time to review the PR :D. I have added it to the bottom of `code.rst` and also I have addressed the suggested changes. Please have a look.

I was aware that I had updated the radare2 submodule, but was not sure whether to add it in this PR or not :sweat_smile:

ITAYC0HEN commented on docs/source/code.rst:
```diff
@@ -177,3 +177,40 @@ Functions documentation
+the radare2 submodule or others, in order to push the most recent
+version of them to Cutter.
+
+Make sure to update the submodule whenever the submodule upstream is updated.
```
```suggestion
```

ITAYC0HEN commented:
Thank you! It looks fantastic :)
Great work, @dhaval0894 !

dhaval0894 commented:
Thanks @ITAYC0HEN!:)
